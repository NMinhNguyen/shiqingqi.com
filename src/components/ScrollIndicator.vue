<template functional>
  <div class="scroll-indicator">
    <div class="scroll-indicator__arrow"></div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss">
@keyframes phase {
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0.2;
  }
}

.scroll-indicator {
  position: fixed;
  z-index: $z-sink;
  left: 0;
  right: 0;
  bottom: 4rem;
  display: flex;
  justify-content: center;
  opacity: 0.3;

  &__arrow {
    position: relative;
    opacity: 0.2;
    animation: phase 4s ease infinite;

    &:before,
    &:after {
      content: "";
      position: absolute;
      display: block;
      width: 0.5rem;
      height: 1.8rem;
      background-color: $on-primary;
      transform-origin: 50% 90%;
    }

    &:before {
      transform: rotate(50deg);
    }

    &:after {
      transform: rotate(-50deg);
    }
  }
}
</style>
