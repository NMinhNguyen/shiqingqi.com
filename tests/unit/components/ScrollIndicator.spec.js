import { shallowMount } from "@vue/test-utils";
import ScrollIndicator from "@/components/ScrollIndicator.vue";

describe("ScrollIndicator", () => {
  it("renders arrows", () => {
    expect(shallowMount(ScrollIndicator)).toMatchSnapshot();
  });
});
