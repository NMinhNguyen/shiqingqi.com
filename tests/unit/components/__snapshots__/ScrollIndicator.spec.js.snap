// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollIndicator renders arrows 1`] = `
<div class="scroll-indicator">
  <div class="scroll-indicator__arrow"></div>
</div>
`;
